File: de.unistuttgart.iste.modeling.generator/src-framework/de/unistuttgart/hamster/framework/FrameworkAssert.java
package de.unistuttgart.hamster.framework;

public class FrameworkAssert {

	private FrameworkAssert() {}
	
	public static void assertAttributeCondition(String condition, boolean valid) {
		if (!valid) {
			throw new RuntimeException("Attribute condition is not valid: " + condition);
		}
	}
	
	public static void assertInstanceNotNull(String instanceName, Object instance) {
		if (instance == null) {
			throw new RuntimeException("Instance must not be null: " + instanceName);
		}
	}
	
	public static void assertObjectFound(String referenceName, String objectTypeName, Object instance) {
		if (instance == null) {
			throw new RuntimeException("Object of type '" + objectTypeName + "' for reference '" + referenceName + "' not found");	
		}
	}

}

File: de.unistuttgart.iste.sqa.mpw.hamstersimulator.cpp/core/src/framework/FrameworkAssert.cpp
#include "FrameworkAssert.h"

#include <string>
#include "AssertionFailedException.h"

namespace framework {
  void assertAttributeCondition(const std::string& condition, bool valid)
  {
    if (!valid) {
      std::string message = "Attribute condition is not valid: " + condition;
      throw AssertionFailedException(message.c_str());
    }
  }

  void assertInstanceNotNull(const std::string& instanceName, EntityReference instance)
  {
    if (instance == nullptr) {
      std::string message = "Instance must not be null: " + instanceName;
      throw AssertionFailedException(message.c_str());
    }
  }

  void assertObjectFound(const std::string& referenceName, const std::string& objectTypeName,
    EntityReference instance)
  {
    if (instance == nullptr) {
      std::string message = "Object of type '" + objectTypeName + "' for reference '" + referenceName + "' not found";
      throw AssertionFailedException(message.c_str());
    }
  }
}
